$OpenBSD: patch-src_FbTk_App_cc,v 1.1 2011/12/29 16:15:53 dcoppa Exp $

Make startup failure easier to diagnose
(upstream git commit afa870802ba4296a4ce8c2b9185250f8da1124ca)

--- src/FbTk/App.cc.orig	Sat Oct 29 15:46:15 2011
+++ src/FbTk/App.cc	Thu Dec 29 16:41:30 2011
@@ -60,8 +60,13 @@ App::App(const char *displayname):m_done(false), m_dis
     if (displayname != 0 && displayname[0] == '\0')
         displayname = 0;
     m_display = XOpenDisplay(displayname);
-    if (!m_display)
-        throw std::string("Couldn't connect to XServer");
+    if (!m_display) {
+        if (displayname) {
+            throw std::string("Couldn't connect to XServer") + displayname;
+        } else {
+            throw std::string("Couldn't connect to XServer passing null display");
+        }
+    }
 
     FbStringUtil::init();
 }
